src/electos/ballotmaker/data/extractor.py

```diff
@@ -0,0 +1,311 @@
+from typing import Dict, List, Union
+
+from electos.ballotmaker.data.models import ElectionData
+from electos.datamodels.nist.indexes import ElementIndex
+from electos.datamodels.nist.models.edf import (
+    BallotMeasureContest,
+    BallotMeasureSelection,
+    BallotStyle,
+    Candidate,
+    CandidateContest,
+    CandidateSelection,
+    Election,
+    ElectionReport,
+    InternationalizedText,
+    OrderedContest,
+    OrderedHeader,
+)
+
+
+# --- Base Types
+#
+# Schema expresses these as union types not subclasses
+
+Contest = Union[BallotMeasureContest, CandidateContest]
+
+OrderedContent = Union[OrderedContest, OrderedHeader]
+
+
+# --- Utilities
```

Comment:
I labelled those "utility" functions because they are more general than the ones in the extractor and are candidates for being moved up into e.g. `nist-datamodels` as utility functions. Unpacking `InternationalizeText` happens over and over, and walking `OrderedContent/OrderedHeader`s is something we'll want to use elsewhere too.

By contrast most of the other functions are specific to BallotMaker's data models.

Reply:
OK, that makes sense. I figured there was a reason.